### PR TITLE
Add additional OCaml versions to GHA CICD

### PR DESCRIPTION
**Pull request by rivy**
- additional CICD OCaml versions/variants to be build by CICD workflow (by request; see <https://github.com/bcpierce00/unison/pull/388#issuecomment-707397387>)

### CODE REVIEW

File: .github/workflows/workflow.yml [hunk 1]
```diff
@@ -15,20 +15,26 @@ jobs:
...
         - { os: macos-latest   , ocaml-version: 4.08.1 }
         - { os: macos-latest   , ocaml-version: 4.07.1 }
+        - { os: macos-latest   , ocaml-version: 4.05.0 }
+        - { os: ubuntu-latest  , ocaml-version: 4.11.1 }
         - { os: ubuntu-latest  , ocaml-version: 4.10.1 }
         - { os: ubuntu-latest  , ocaml-version: 4.10.0+musl+static+flambda }
         - { os: ubuntu-latest  , ocaml-version: 4.09.1 }
         - { os: ubuntu-latest  , ocaml-version: 4.08.1 }
         - { os: ubuntu-latest  , ocaml-version: 4.07.1 }
-        - { os: windows-latest , ocaml-version: 4.10.1+mingw64 }
```
Comment (gdt) [inline .github/workflows/workflow.yml:27]: Why is this deleted?

Comment (rivy) [reply .github/workflows/workflow.yml:27]: The Windows OCaml installation defaults to the MinGW 64-bit version, so, "4.10.1+mingw64" is the same as just "4.10.1".

**Comment by rivy** [comment]
The failures look like a transient network error when downloading the ubuntu-18 musl-tools fix (see <https://github.com/ocaml/ocaml/issues/9131#issuecomment-599765888>) during the OCaml setup. If it recurs with any frequency, `ubuntu-20.04` can be used instead of `ubuntu-latest`.

**Comment by gdt** [comment]
OK - we can think about transient errors and versions later.
